### PR TITLE
PNDA-4431: Add basic platform test for Flink in PNDA

### PR DESCRIPTION
**Pull request by GaneshManal**
# Problem Statement:
PNDA-4431: Add basic platform test for Flink in PNDA

# Change:
Added the flink plugin to platform testing which sends api request to the flink history server.
APIS being called -
- config - on success, reads flink version installed and add it to health matrix.
- joboverview - on success, adds the number of completed jobs to health matrix.

# Test details:
Verified the fix for AWS:
- UBUNTU - PICO -CDH & HDP
- UBUNTU - STD -CDH & HDP
- RHEL - PICO -CDH & HDP
- RHEL - STD -CDH & HDP
- Verification Done for openstack.

### CODE REVIEW

File: src/main/resources/plugins/flink/TestbotPlugin.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    def __init__(self):
+        pass
+
+    def read_args(self, args):
+        '''
+        This class argument parser.
+        This shall come from main runner in the extra arg
+        '''
+        parser = argparse.ArgumentParser(prog=self.__class__.__name__,
+                                         usage='%(prog)s [options]', description='Key metrics from CDH cluster')
```
Comment (jgarnier) [inline src/main/resources/plugins/flink/TestbotPlugin.py]: replace 'Key metrics from CDH cluster' by 'Show state of Flink History server'

**Review by jgarnier — CHANGES_REQUESTED** [review]

**Comment by jgarnier** [comment]
could you also review the PYLINT output in order to improve code quality please?

**Comment by GaneshManal** [comment]
Testbotplugin.py - pylint score = 9.26/10

File: src/main/resources/plugins/flink/TestbotPlugin.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+Purpose:      Blackbox test for the Deployment manager
+
+"""
+
+import time
+import argparse
+import json
+import eventlet
+
+import requests
```
Comment (jgarnier) [inline src/main/resources/plugins/flink/TestbotPlugin.py:25]: requests need to be imported before json and also group it with requests.exceptions

File: src/main/resources/plugins/flink/TestbotPlugin.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        """
+        plugin_args = args.split() \
+        if args is not None and args.strip() \
+        else ""
+
+        options = self.read_args(plugin_args)
+        cause = []
+        values = []
+
+        history_server_available_success, history_server_completed_jobs_success = False, False
```
Comment (jgarnier) [inline src/main/resources/plugins/flink/TestbotPlugin.py]: too long variable names, could you replace history_server_ by hs_ please

**Review by jgarnier — CHANGES_REQUESTED** [review]

**Pull request merged**